rust/arrow/src/datatypes.rs

```diff
@@ -121,17 +122,71 @@ pub trait ArrowPrimitiveType: 'static {
     fn default_value() -> Self::Native;
 }
 
-impl ArrowNativeType for bool {}
-impl ArrowNativeType for i8 {}
-impl ArrowNativeType for i16 {}
-impl ArrowNativeType for i32 {}
-impl ArrowNativeType for i64 {}
-impl ArrowNativeType for u8 {}
-impl ArrowNativeType for u16 {}
-impl ArrowNativeType for u32 {}
-impl ArrowNativeType for u64 {}
-impl ArrowNativeType for f32 {}
-impl ArrowNativeType for f64 {}
+impl ArrowNativeType for bool {
+    fn into_json_value(self) -> Option<Value> {
+        Some(self.into())
+    }
+}
+
+impl ArrowNativeType for i8 {
+    fn into_json_value(self) -> Option<Value> {
+        Some(VNumber(Number::from(self)))
+    }
+}
+
+impl ArrowNativeType for i16 {
+    fn into_json_value(self) -> Option<Value> {
+        Some(VNumber(Number::from(self)))
+    }
+}
+
+impl ArrowNativeType for i32 {
+    fn into_json_value(self) -> Option<Value> {
+        Some(VNumber(Number::from(self)))
+    }
+}
+
+impl ArrowNativeType for i64 {
```

Comment:
Is it fine that we implement this on `ArrowNativeType` (for i{32|64} which can also be dates and times)? Should we support temporal arrays? I know with time, we don't really have a JSON format, but we could have JSON timestamps that we might benefit from comparing.

Reply:
Do you mean `ArrowPrimitiveType`? In fact I don't think so, because temporal arrays already benefits from this because their native type is i32|i64.

Reply:
Yes, what I was considering was how we'd consider dates in JSON, as they're normally in the string format of `2019-07-29T05:07:07.771Z`. I think it's not an issue though, so you may disregard my comment :)

Reply:
The challenging part is that primitive array's value method returns T::Native. I think we can still compare temporal type if we serialize date time type to time stamp in json.